app/trace/trace_viewer.tsx

```diff
@@ -21,28 +21,48 @@ export interface TraceViewProps {
   filterHidden?: boolean;
 }
 
+interface TraceViewerState {
+  filter: string;
+  totalMatches: number;
+  currentMatch: number; // 1-based index of the currently highlighted match
+}
+
 // The browser starts struggling if we have a div much greater than this width
 // in pixels. For now we rely on the browser for rendering the horizontal
 // scrollbar, so we don't allow the horizontally scrollable width to exceed this
 // value.
 const SCROLL_WIDTH_LIMIT = 18_000_000;
 
+// When jumping to a matched span via the search bar, ensure that the span is
+// large enough on screen so it can be seen and interacted with. If the span is
+// smaller than this in pixel width, we zoom in to make it at least this wide.
+// Conversely, if the span is very large, we attempt to zoom out (down to the
+// "normal" min-zoom level) so that more context is visible.
+const MIN_SPAN_PIXEL_WIDTH = 1;
```

Comment:
nit: we probably want to reference the same value that is used to decide whether a span is allowed to render or not. Or if there isn't a constant, probably worth adding one. LGTM otherwise